src/util/JoinAlgorithms/JoinAlgorithms.h

```diff
@@ -150,7 +152,9 @@ CPP_template(typename Range1, typename Range2, typename LessThan,
                                                        outOfOrderFound)) {
         if (lessThan(*it, *itFromRight)) {
           compatibleRowAction(it, itFromRight);
-          cover(it);
+          if constexpr (CoverUndefRanges::value) {
```

Comment:
Doesn't std::true type implcitly convert to bool and thus the `::value`is redundant?

Reply:
The instance of the type yes, but here I'm taking the type and turning it into a value.